src/main/java/seedu/address/model/person/Doctor.java

```diff
@@ -29,4 +30,67 @@ public Specialty getSpecialty() {
     public Yoe getYoe() {
         return yoe;
     }
+
+    /**
+     * Returns true if both doctors have the same name.
+     * This defines a weaker notion of equality between two doctors.
+     */
+    public boolean isSameDoctor(Doctor otherDoctor) {
+        if (otherDoctor == this) {
+            return true;
+        }
+
+        return otherDoctor != null
+                && otherDoctor.getName().equals(getName());
+    }
+
+    /**
+     * Returns true if both doctors have the same identity and data fields.
+     * This defines a stronger notion of equality between two doctors.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Doctor)) {
+            return false;
+        }
+
+        Doctor otherDoctor = (Doctor) other;
+        return otherDoctor.getName().equals(getName())
+                && otherDoctor.getPhone().equals(getPhone())
+                && otherDoctor.getEmail().equals(getEmail())
+                && otherDoctor.getSpecialty().equals(getSpecialty())
+                && otherDoctor.getYoe().equals(getYoe())
+                && otherDoctor.getTags().equals(getTags());
+    }
+
+    @Override
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(getName(), getPhone(), getEmail(), getSpecialty(), getYoe(), getTags());
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(getName())
+                .append("; Phone: ")
+                .append(getPhone())
+                .append("; Email: ")
+                .append(getEmail())
+                .append("; Specialty: ")
```

Comment:
```suggestion
.append("; Speciality: ")
```
Standardise spelling across Docedex